internal/unsecure/client.go

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ * Copyright 2019 VMWare Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+package unsecure
+
+import (
+	"flag"
+	"fmt"
+	"github.com/BurntSushi/toml"
+	"github.com/globalsign/mgo"
+	"os"
+
+	"github.com/edgexfoundry/docker-edgex-mongo/internal/pkg"
+)
+
+type unsecureConfigurationStruct struct {
+	pkg.BaseConfigurationStruct
+	Credentials map[string]pkg.CredentialsInfo
+}
+
+var databaseCollectionsMap = map[string]func(db *mgo.Database){
+	//TODO Diana: Do I need empty database ?!
+	"authorization":   nil,
+	"admin":           creatAdminCollections,
+	"metadata":        createMetadataCollections,
+	"coredata":        createCoredataCollections,
+	"rules_engine_db": nil,
+	"notifications":   createNotificationCollections,
+	"scheduler":       createSchedulerCollections,
+	"logging":         createLoggingCollections,
+	"exportclient":    createExpertClientCollections,
+}
+
+type unsecureClient struct {
+	Configuration *unsecureConfigurationStruct
+}
+
+func NewUnsecureClient() (*unsecureClient, error) {
+	config, err := LoadConfig("res/configuration.toml")
+	if err != nil {
+		return nil, err
+	}
+	client := unsecureClient{
+		Configuration: config,
+	}
+
+	return &client, nil
+}
+
+func (client *unsecureClient) PopulateDatabase() {
+	session, err := pkg.GetSession(&client.Configuration.BaseConfigurationStruct)
+	if err != nil {
+		pkg.LoggingClient.Error(fmt.Sprintf("%s failed because of %s", pkg.EdgexMongoServiceKey, err.Error()))
+		os.Exit(1)
+	}
+
+	defer session.Close()
```

Comment:
Be advised that this `defer` will not run if the `os.Exit(1)` above is called.
https://gobyexample.com/exit

Reply:
Thank you for the comment.. I have rewritten the app - removing the abstraction and factory method taking in account this comment.